apps/web/pages/availability/[schedule].tsx

```diff
@@ -177,26 +178,43 @@ const SmallScreenSideBar = ({ open, children }: { open: boolean; children: JSX.E
     </div>
   );
 };
+
 export default function Availability() {
   const searchParams = useCompatSearchParams();
-  const { t, i18n } = useLocale();
-  const utils = trpc.useUtils();
-  const me = useMeQuery();
   const scheduleId = searchParams?.get("schedule") ? Number(searchParams.get("schedule")) : -1;
-  const fromEventType = searchParams?.get("fromEventType");
-  const { timeFormat } = me.data || { timeFormat: null };
-  const [openSidebar, setOpenSidebar] = useState(false);
   const { data: schedule, isPending } = trpc.viewer.availability.schedule.get.useQuery(
     { scheduleId },
     {
       enabled: !!scheduleId,
     }
   );
 
+  // TODO: reimplement Skeletons for this page in here
+  if (isPending) return null;
+
+  if (!schedule) return null;
+
+  // We wait for the schedule to be loaded before rendering the form since `defaultValues`
+  // cannot be redeclared after first render. And using `values` will trigger a form reset
+  // when revalidating.
+  return <AvailabilityForm schedule={schedule} isPending={isPending} />;
+}
+
+type ScheduleProp = RouterOutputs["viewer"]["availability"]["schedule"]["get"];
+function AvailabilityForm({ schedule, isPending }: { schedule: ScheduleProp; isPending: boolean }) {
+  const searchParams = useCompatSearchParams();
+  const { t, i18n } = useLocale();
+  const utils = trpc.useUtils();
+  const me = useMeQuery();
+  const scheduleId = searchParams?.get("schedule") ? Number(searchParams.get("schedule")) : -1;
+  const fromEventType = searchParams?.get("fromEventType");
+  const { timeFormat } = me.data || { timeFormat: null };
+  const [openSidebar, setOpenSidebar] = useState(false);
+
   const form = useForm<AvailabilityFormValues>({
-    values: schedule && {
+    defaultValues: {
```

Comment:
If we're going to use values. At least we should use `resetOptions: { keepDirtyValues: true }` as stated in [the docs](https://react-hook-form.com/faqs#Howtoinitializeformvalues). But IMO is better to ensure initial data is always defined when initializing the form to avoid side effects.